lib/cookie/cookie.ts

```diff
@@ -382,46 +393,42 @@ const cookieDefaults = {
   creation: null,
   lastAccessed: null,
   sameSite: undefined,
-}
-
-type CreateCookieOptions = {
-  key?: string
-  value?: string
-  expires?: Date | 'Infinity' | null
-  maxAge?: number | 'Infinity' | '-Infinity'
-  domain?: string | null
-  path?: string | null
-  secure?: boolean
-  httpOnly?: boolean
-  extensions?: string[] | null
-  creation?: Date | 'Infinity' | null
-  creationIndex?: number
-  hostOnly?: boolean | null
-  pathIsDefault?: boolean | null
-  lastAccessed?: Date | 'Infinity' | null
-  sameSite?: string | undefined
-}
+} as const satisfies Required<CreateCookieOptions>
```

Comment:
* `as const` makes the defaults a readonly value and uses the most precise type possible, e.g .`""` instead of `string`
* `satisfies Require<CreateCookieOptions>` ensures that we adhere to that interface without losing the precision of the type we already have, e.g. we keep `"Infinity"` for `expires`, rather than getting `Date | "Infinity" | null` from `CreateCookieOptions`